### ISSUE TITLE
CI should run correctly if PRs are submitted to the origin repository

### ISSUE THREAD

**Issue by mihaibudiu** [feldera/feldera#627]
Everyone else accepts contributions from the outside by asking contributors to fork and submit to the origin repository.
We ask PRs to be sent to upstream.
That is not scalable, and it won't work for external contributors.
We should fix this if we expect people to contribute, even if it means running fewer tests for external contributions.

**Comment by blp** [comment]
I found the wording in this issue to be difficult to understand. Let me rephrase it.

The usual thing on github for submitting a contribution is to fork the upstream repo, push a branch to the fork, and submit a PR from the fork to the upstream repository. We instead ask contributors to do something else, which isn't scalable and won't work externally.

**Comment by lalithsuresh** [comment]
What happens right now when someone sends us a PR from a fork? We should run the CI workflows then.

We absolutely should not be running the CI for commits to a forked repository (which is what the wording of this PR's title suggests).

**Comment by mihaibudiu** [comment]
Every other repository I contribute to does run CI for PRs from forks, but sometimes it asks a maintainer to approve them.
![image](https://github.com/feldera/feldera/assets/19397042/44153bd2-c884-4843-a7cf-de0f2418b4db)

**Comment by mihaibudiu** [comment]
For Calcite it always runs them.

**Comment by mihaibudiu** [comment]
But Calcite deliberately keeps the CI tests to a relatively small size. Which is probably why Calcite is insufficiently tested.

**Comment by lalithsuresh** [comment]
What's the behavior on our repo right now? That we don't run the CI at all for PRs from forks?

**Comment by mihaibudiu** [comment]
we run and fail

**Comment by gz** [comment]
I'm not sure it's true. My understanding is:

a) If you push to a branch on your account, the CI will currently fail. That's because it doesn't find the runner type (our own machines) on your account (which we specify in the github yaml).
b) If you were to make a PR from your branch to feldera/feldera we would be able to approve it (like the screenshot that was posted above) which would then run in our environment.

I think b) is probably ok the way it is (if that statement is true -- we should test it). 
And a) is what the issue is complaining about. Then, we would need a fix for a) which might involve just picking the default github runner in this case. 

Does this make sense?

**Comment by gz** [comment]
For some better explanation: we set this in the configuration (which avoids running arbitrary PRs without our approval):

![Screenshot from 2023-09-06 23-41-44](https://github.com/feldera/feldera/assets/127654/1bb45c32-9629-454c-bb61-07410db3fcd3)

And then we also have our own runners:
![Screenshot from 2023-09-06 23-43-04](https://github.com/feldera/feldera/assets/127654/ab0f5497-a007-4a6f-8762-8075b56ce671)

Which we request to run on in our ci.yml file: https://github.com/feldera/feldera/blob/main/.github/workflows/ci.yml#L32
In your local repo this runner usually doesn't exist, so it will fail (after 24h of not finding the runner). But it doesn't mean that we can't run PRs that are submitted from external repos in our own runners (after our approval).

**Comment by lalithsuresh** [comment]
Stale issue. Closing for now.
